aws/onboarding/byok8s/python/rubrik_eks_cluster_private.py

```diff
@@ -0,0 +1,1203 @@
+        raise
+
+
+def strip_paths_from_arn(arn: str) -> str:
+    """Strip paths from ARN
+
+    Args:
+    Returns:
+        returns ARN without paths
+    """
+    try:
+        parsed_arn = arnparse.arnparse(arn)
+        resource_parts = str.split(parsed_arn.resource, "/")
+        if len(resource_parts) <= 0:
+            raise Exception("Invalid ARN")
+
+        # While constructing the aws-auth configmap, the master/worker node role ARN
+        # should not include paths.
+        # (Ref: https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html)
+        role_without_path = resource_parts[-1]
+        arn_without_path = (
+            f"{ARN_PREFIX}{parsed_arn.partition}{ARN_DELIMITER}"
+            f"{parsed_arn.service}{ARN_DELIMITER}{ARN_DELIMITER}"
+            f"{parsed_arn.account_id}{ARN_DELIMITER}{parsed_arn.resource_type}/"
+            f"{role_without_path}"
+        )
+        return arn_without_path
+    except Exception as e:
+        print(f"Error stripping paths from ARN: {str(e)}")
+        raise
+
+
+def persist_kubectl_configuration(
+        cluster_endpoint: str,
+        cluster_certificate_authority: str,
+        cluster_name: str,
+        cluster_arn: str
+) -> str:
+    """Persist kubectl configuration to a file for kubectl commands.
+
+    Args:
+        cluster_endpoint: cluster endpoint
+        cluster_certificate_authority: cluster certificate authority
+        cluster_name: cluster name
+        cluster_arn: cluster ARN
+    Returns:
+        returns cluster config file path
+    """
+    try:
+        cluster_config = {
+            "apiVersion":
+                "v1",
+            "kind":
+                "Config",
+            "clusters": [{
+                "cluster": {
+                    "server":
+                        str(cluster_endpoint),
+                    "certificate-authority-data":
+                        str(cluster_certificate_authority)
+                },
+                "name": cluster_arn
+            }],
+            "contexts": [{
+                "context": {
+                    "cluster": cluster_arn,
+                    "user": cluster_arn,
+                },
+                "name": cluster_arn
+            }],
+            "current-context":
+                cluster_arn,
+            "preferences": {},
+            "users": [{
+                "name": cluster_arn,
+                "user": {
+                    "exec": {
+                        "apiVersion": "client.authentication.k8s.io/v1alpha1",
+                        "command": "aws-iam-authenticator",
+                        "args": ["token", "-i", cluster_name]
+                    }
+                }
+            }]
+        }
+
+        config_text = yaml.dump(cluster_config, default_flow_style=False)
+
+        # Determine the path for the kubectl config file
+        kube_dir = expanduser("~") + "/.kube"
+        config_file = kube_dir + "/config"
```

Comment:
this is a common path, a customer may already have some kubeconfigs on this file. Should we instead create a file with a random identifier instead, ex /.kube-{random-id} etc

Reply:
We cannot store it into some different path, this is the place where Cluster Configuration, Context management, etc config is stored. when we run the kubectl commands, config values is picked from here.
This script is meant to run on the new Jumpbox asaik, @RudraNirvan can you please comment on this

Reply:
We can store it in different path? When setting up the kube APIs it takes the config file path as an input

Reply:
If we store in the different path, we won't be able to run any kubectl command on the jumpbox, as the context and other config would be missing. This is the path where kubectl contexts and token resides

Reply:
Given:
1. The additional effort needed to support that.
2. We not knowing how the customers are going to use this script.
3. There is a way to switch to use a different kube config.

We need to change this for now?

Reply:
Makes sense to keep as is, specially given the how the customer will use it is it known.

Reply:
Makes sense to keep as is, specially given the how the customer will use it is it known.